### PR TITLE
PDF render in the documentation generators #252

### PR DESCRIPTION
**Pull request by diarraa**

### CODE REVIEW

File: miscellaneous/roboconf-doc-generator/src/main/java/net/roboconf/doc/generator/internal/renderers/PdfRenderer.java [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+				FopFactory fopFactory = FopFactory.newInstance( fopConfig );
+				Fop fop =  fopFactory.newFop( "application/pdf", out );
+				Source src = new StreamSource( index_fo );
+
+				TransformerFactory factory = TransformerFactory.newInstance();
+				Transformer transformer = factory.newTransformer();
+
+				Result res = new SAXResult(fop.getDefaultHandler());
+				transformer.transform(src, res);
+		} catch (SAXException e) {
```
Comment (vincent-zurczak) [inline miscellaneous/roboconf-doc-generator/src/main/java/net/roboconf/doc/generator/internal/renderers/PdfRenderer.java]: Wrap all your catch blocks within a single one.
And propagate them as an IO exception.

```java
try {

} catch( Exception e) {
         throw new IOException( e );

} finally {

}
```

File: miscellaneous/roboconf-doc-generator/src/main/java/net/roboconf/doc/generator/internal/renderers/PdfRenderer.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+	}
+
+	@Override
+	protected String renderTitle3(String title) {
+		return "";
+	}
+
+	@Override
+	protected String renderParagraph(String paragraph) {
+		//return this.fopfile.renderParagraph( paragraph );
```
Comment (vincent-zurczak) [inline miscellaneous/roboconf-doc-generator/src/main/java/net/roboconf/doc/generator/internal/renderers/PdfRenderer.java]: Remove useless comments.

File: miscellaneous/roboconf-doc-generator/src/main/java/net/roboconf/doc/generator/internal/renderers/PdfRenderer.java [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+	}
+
+	@Override
+	protected String renderPageBreak() {
+		return "";
+	}
+
+	@Override
+	protected String indent() {
+		//return this.fopfile.indent();
```
Comment (vincent-zurczak) [inline miscellaneous/roboconf-doc-generator/src/main/java/net/roboconf/doc/generator/internal/renderers/PdfRenderer.java]: Remove useless comments.

File: miscellaneous/roboconf-doc-generator/src/main/java/net/roboconf/doc/generator/internal/renderers/PdfRenderer.java [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+	}
+
+	@Override
+	protected String addTableLine(String... lineEntries) {
+		return "";
+	}
+
+	@Override
+	protected String renderDocumentTitle() {
+		//return this.fopfile.renderDocumentTitle();
```
Comment (vincent-zurczak) [inline miscellaneous/roboconf-doc-generator/src/main/java/net/roboconf/doc/generator/internal/renderers/PdfRenderer.java]: Remove useless comments.

**Comment by vincent-zurczak** [comment]
You should also remove test resources related to PDF.

**Comment by vincent-zurczak** [comment]
It seems also Github will not accept to merge this PR directly as it indicates there are conflicts. :fearful: 
You will have to squash by hand and then fetch/merge with the master before we can merge the PR. :mask:

**Comment by diarraa** [comment]
Files related to PDF test are stored in  outputDir. This directory is removed after the test.

**Comment by vincent-zurczak** [comment]
Indeed. But there is a **fop.xconf** file.
And I think I saw a similar file under **src/main/java**.

**Comment by vincent-zurczak** [comment]
I close this PR.
Please, make a new one.

**Pull request closed without merging**